src/status_im/chat/models/message.cljs

```diff
@@ -1,214 +1,186 @@
+          {:db db}
+          chat-ids))
+
+(fx/defn add-senders-to-chat-users
+  {:events [:chat/add-senders-to-chat-users]}
+  [{:keys [db]} messages]
+  (reduce (fn [acc {:keys [chat-id alias name identicon from]}]
+            (update-in acc [:db :chats chat-id :users] assoc
+                       from
+                       (mentions/add-searchable-phrases
+                        {:alias      alias
+                         :name       (or name alias)
+                         :identicon  identicon
+                         :public-key from
+                         :nickname   (get-in db [:contacts/contacts from :nickname])})))
+          {:db db}
+          messages))
+
+(defn timeline-message? [db chat-id]
+  (and
+   (get-in db [:pagination-info constants/timeline-chat-id :messages-initialized?])
+   (when-let [pub-key (get-in db [:chats chat-id :profile-public-key])]
+     (contact.db/added? db pub-key))))
+
+(defn get-timeline-message [db chat-id message-js]
+  (when (timeline-message? db chat-id)
+    (data-store.messages/<-rpc (types/js->clj message-js))))
+
+(defn add-message [{:keys [db] :as acc} message-js chat-id message-id cursor-clock-value]
+  (let [{:keys [alias replace from clock-value] :as message}
+        (data-store.messages/<-rpc (types/js->clj message-js))]
+    (if (message-loaded? db chat-id message-id)
+      ;; If the message is already loaded, it means it's an update, that
+      ;; happens when a message that was missing a reply had the reply
+      ;; coming through, in which case we just insert the new message
+      (assoc-in acc [:db :messages chat-id message-id] message)
+      (cond-> acc
+        ;;add new message to db
+        :always
+        (update-in [:db :messages chat-id] assoc message-id message)
+        :always
+        (update-in [:db :message-lists chat-id] message-list/add message)
+
+        (or (not cursor-clock-value) (< clock-value cursor-clock-value))
+        (update-in [:db :pagination-info chat-id] assoc
+                   :cursor (chat.loading/clock-value->cursor clock-value)
+                   :cursor-clock-value clock-value)
+
+        ;;conj sender for add-sender-to-chat-users
+        (and (not (string/blank? alias))
+             (not (get-in db [:chats chat-id :users from])))
+        (update :senders assoc from message)
+
+        (not (string/blank? replace))
+        ;;TODO this is expensive
+        (hide-message chat-id replace)))))
+
+(defn reduce-js-messages [{:keys [db] :as acc} ^js message-js]
+  (let [chat-id (.-localChatId message-js)
+        clock-value (.-clock message-js)
+        message-id (.-id message-js)
+        current-chat-id (:current-chat-id db)
+        cursor-clock-value (get-in db [:pagination-info current-chat-id :cursor-clock-value])]
+    ;;ignore not opened chats and earlier clock
+    (if (and (get-in db [:pagination-info chat-id :messages-initialized?])
+             ;;TODO why do we need this ?
```

Comment:
mmh, I think this might actually not be needed anymore, the check is done in status-go